### PR TITLE
Demonstrate make_array helper

### PR DESCRIPTION
**Pull request by OwenArnold**
Since support for this language feature has come so late to clang (see comments), I recommend this small addition to give better compiler support for most users/developers.

### CODE REVIEW

File: src/tag_util.h [hunk 1]
```diff
@@ -8,9 +8,30 @@
...
+ * https://clang.llvm.org/cxx_status.html
+ *
+ * This make helper uses function template argument deduction to solve
+ * for std::array declarations.
+ */
+template <typename... T>
+constexpr auto make_array(T&&... values) ->
+    std::array<
+       typename std::decay<
+           typename std::common_type<T...>::type>::type,
```
Comment (SimonHeybrock) [inline src/tag_util.h]: Use C++14's `std::decay_t` and `std::common_type_t` to avoid the `typename ... :: type` verbosity.

Comment (OwenArnold) [reply src/tag_util.h]: 👍

**Comment by SimonHeybrock** [comment]
@OwenArnold With the `..._t` helper the `typename` is also redundant.

The trailing return type can also be omitted in this case (with C++14).

Please check if OSX accepts my update to the branch.

**Comment by OwenArnold** [comment]
@SimonHeybrock all confirmed to work on mac with LLVM/Clang 6.0

**Pull request merged**